### DISCUSSION TITLE
Getting into it this library (documentation feedback)

### DISCUSSION THREAD

**Discussion by JeffreyArts** [donalffons/opencascade.js#285] [General]
Hi,

I am looking for something that would allow me to do some CAD operations in the browser. In this quest I came across this library and it looks very interesting! I wanted to share you my experience as someone who just comes across this module for the first time, since seemed quite challenging to get a demo up and running via the "[get started](https://ocjs.org/docs/getting-started/hello-world)" page.

On this page I miss an overview of how the different component relate to each other (and which components are there in play). The page itself doesn't really explain how to set up a project and the examples just go into how modelling works. The demo code itself on this page seems like it is written for VUE. While the page says that the library is framework agnostic, there are no vanilla JS examples on the page to describe how it should work without the use of any JS frameworks. Also it seems to be required to use a bundler to use the library, which is being explained how to do with Webpack but not with alternative popular bundlers like Vite, Parcel or EsBuild. The developer documentation is just a schematic of how a Docker container with opencascade looks like, and it's unclear how to get insight in what methods there are available. 

This makes it all a quite high barrier for newcomers like myself to get our hands dirty with this library. 

**edit**
In second glance I noticed these things in the header menu, but they were not mentioned in the "getting started" section of the documentation. There is no starter template just for using it without any frameworks.

I've got a workable demo now for myself that allows me to get started exploring this library. But I wanted to provide you my experience nonetheless, as a new user who's looking for a JS web-based CAD tool, whose not familiar with opencascade (nor its community)

**Comment by cartoonshow57** [comment]
Hello! I am struggling with setting up the Hello World example for OpenCascade.js. Can you help me out, please?

**Comment by cartoonshow57** [comment]
I spun up an app using vite and react. My folder structure looks like this:
![Screenshot from 2024-12-15 18-59-33](https://github.com/user-attachments/assets/c132d931-b656-4722-8114-5dc1786a9ee9)

Where should I put the files 'visualize.js' and 'index.js'(These two files are there in the Getting Started Page). And how should the code look like?

**Comment by abrahamshepher** [comment]
I'm having issue in vite react setup

**Reply by tetron432** [reply]
Did you manage to figure it out?
